File: matlab/oecd/oecd_min.m
% Descargador OECD SDMX CSV — script mínimo
% ----------------------------------------------------------------------------
% Objetivo
%   Construir la URL SDMX de OECD, descargar el CSV, cargarlo como tabla y exportar a CSV.
% Parámetros editables
%   - agency_identifier, dataset_identifier, dataset_version (si aplica)
%   - data_selection: clave SDMX tras la '/'
% Salida
%   - Archivo CSV con los datos del dataset seleccionado
% Pasos
%   1) Construir data_identifier agency,dataset,version
%   2) Leer URL directamente a tabla
%   3) Exportar a CSV

% Parámetros básicos
agency_identifier = 'OECD.ECO.MAD';
dataset_identifier = 'DSD_EO@DF_EO';
dataset_version = '';
data_selection = '...'; % clave SDMX específica

% Construir la URL SDMX
base_url = 'https://sdmx.oecd.org/public/rest/data';
data_identifier = [agency_identifier ',' dataset_identifier ',' dataset_version];
url = sprintf('%s/%s/%s', base_url, data_identifier, data_selection);

% Leer la URL directamente a tabla
T = readtable(url, 'FileType', 'text');

% Exportar la tabla a CSV
writetable(T, 'oecd_min.csv', 'FileType', 'text');

File: R/oecd/oecd_min.R
# Descargador mínimo de OECD SDMX (CSV)
#
# Cómo personalizar:
# - Cambie 'data_identifier' para apuntar a {Agencia},{Dataset},{Versión}
# - Cambie 'data_selection' para seleccionar la clave (dimensiones) del dataset
# - Cambie 'params' (startPeriod, endPeriod, dimensionAtObservation)
# - Cambie 'output_name' para definir el nombre del archivo de salida
#
# Sintaxis oficial (OCDE):
# {Host URL}/{Agency identifier},{Dataset identifier},{Dataset version}/{Data selection}?{otros parámetros}

library(httr)

# --- Parámetros editables ---
# Identificadores de dataset
  base_url <- "https://sdmx.oecd.org/public/rest/data" # En principio no hace falta cambiarlo
  agency_identifier <- "OECD.ECO.MAD"
  dataset_identifier <- "DSD_EO@DF_EO"
  dataset_version <- ""
# Construir codigo completo  
  data_identifier <- paste0(agency_identifier, ",", dataset_identifier, ",", dataset_version)
# Rango de fechas y dimensiones
  data_selection <- "FRA+DEU.PDTY.A"
  params <- list(
    startPeriod = "1965",
    endPeriod = "2023",
    dimensionAtObservation = "AllDimensions"
  )
# Nombre del archivo de salida
  output_name <- "oecd_ejemplo"
  out_file <- paste0("data/", output_name, ".csv")

# --- Crear directorio de datos si no existe ---
if (!dir.exists("data")) dir.create("data", recursive = TRUE, showWarnings = FALSE)

# --- Descargar archivo CSV ---
url <- paste0(base_url, "/", data_identifier, "/", data_selection)
res <- httr::GET(url, query = params, httr::accept("text/csv"))
if (httr::http_error(res)) stop(sprintf("OECD request failed [%s]", httr::status_code(res)))
writeBin(httr::content(res, as = "raw"), out_file)


